npm/install.js

```diff
@@ -36,23 +36,56 @@ fs.unlink(path.join(installScriptLocation, 'cmf-portal'), (err) => {
     }
 });
 
-// download respective release zip from github
-console.info(`Current platform / arch: ${process.platform} / ${process.arch}`);
-const pkgUrl = `https://github.com/criticalmanufacturing/portal-sdk/releases/download/#{Version}#/Cmf.CustomerPortal.Sdk.Console-${process.env.npm_package_version}.${PLATFORM_MAPPING[process.platform]}-${ARCH_MAPPING[process.arch]}.zip`;// opts.binUrl.replace("{{version}}", opts.version).replace("{{platform}}", PLATFORM_MAPPING[process.platform]).replace("{{arch}}", ARCH_MAPPING[process.arch]);
-console.info(`Getting release archive from ${pkgUrl} into ${path.resolve(installScriptLocation)}`);
-axios.get(pkgUrl, { responseType: 'arraybuffer' })
-     .then(function (response) {
-         // handle success
-         const zip = tmp.tmpNameSync();
-         console.log(`Writing temporary zip file to ${zip}`);
-         fs.writeFileSync(zip, response.data);
-         console.log(`Extracting zip file ${zip} to ${installScriptLocation}`);
-         (new AdmZip(zip)).extractAllTo(installScriptLocation, undefined, true);
-         rimraf.sync(zip)
-     })
-     .catch(function (error) {
-         // handle error
-         console.error(error);
-         console.error(error(`Could not install version ${process.env.npm_package_version} on your platform ${process.platform}/${process.arch}: ${error.message}`));
-         process.exit(1);
-     });
+// Function to download and extract the zip file
+async function downloadAndExtract(pkgUrl, installScriptLocation) {
+    try {
+        console.info(`Fetching release archive from ${pkgUrl}`);
+        const response = await axios.get(pkgUrl, { responseType: 'arraybuffer' });
+        
+        const zip = tmp.tmpNameSync();
+        console.log(`Writing temporary zip file to ${zip}`);
+        fs.writeFileSync(zip, response.data);
+
+        console.log(`Extracting zip file ${zip} to ${installScriptLocation}`);
+        (new AdmZip(zip)).extractAllTo(installScriptLocation, undefined, true);
+        
+        rimraf.sync(zip); 
+    } catch (error) {
+        throw new Error(`Failed to download or extract archive from ${pkgUrl}: ${error.message}`);
+    }
+}
+
+// Main function to attempt downloading from multiple sources
+async function installPackage() {
+    console.info(`Current platform / arch: ${process.platform} / ${process.arch}`);
+    const installScriptLocation = path.resolve(__dirname);  // Adjust as necessary
+
+    // Primary URL
+    
+    const primaryUrl = new URL(`https://github.com/criticalmanufacturing/portal-sdk/releases/download/#{Version}#/Cmf.CustomerPortal.Sdk.Console-${version}.${PLATFORM_MAPPING[process.platform]}-${ARCH_MAPPING[process.arch]}.zip`);
+    
+    // Fallback URLs
+    const fallbackUrls = [
+        new URL(`https://criticalmanufacturing.io/repository/tools/Cmf.CustomerPortal.Sdk.Console-${version}.${PLATFORM_MAPPING[process.platform]}-${ARCH_MAPPING[process.arch]}.zip`),
```

Comment:
All `${version}` findings should be changed to `${process.env.npm_package_version}`.